dali/python/nvidia/dali/plugin/pytorch/proxy/__init__.py

```diff
@@ -0,0 +1,487 @@
+#     |               |                    |                |-------------------------------->|
+#     |               |                    |                |                   |             |
+#     |<--data_0_0----|                    |                |                   |             |
+#     |               |                    |                |                   |             |
+#     |~~~get()~~~~~~>|                    |                |                   |             |
+#     |               |                    |~~~get()~~~~~~~>|                   |             |
+#     |               |                    |                |                   |             |
+#     |               |                    |<--req_1_0------|                   |             |
+#     |               |                    |                |                   |             |
+#     |               |             ---------------         |                   |             |
+#     |               |             | run         |         |                   |             |
+#     |               |             ---------------         |                   |             |
+#     |               |                    |                |                   |             |
+#     |               |                    |                |<~~put(req_0_1)~~~~|             |
+#     |               |                    |                |                   |             |
+#     |               |                    |                |------------------>|             |
+#     |               |                    |                |                   |             |
+#     |               |<~~put(data_1_0)~~~~|                |                   |             |
+#     |               |                    |                |                   |             |
+#     |               |------------------->|                |                   |             |
+#     |               |                    |                |                   |             |
+#     |<--data_1_0----|                    |                |                   |             |
+#     |               |                    |                |                   |             |
+# +-------+   +---------------+     +-------------+ +---------------+     +-----------+ +-----------+
+# | main  |   | dali_output_q |     | data_thread | | dali_input_q  |     | worker_0  | | worker_1  |
+# +-------+   +---------------+     +-------------+ +---------------+     +-----------+ +-----------+
+
+
+def _external_source_node_names(pipeline):
+    """
+    extract the names of all the ExternalSource nodes in the pipeline
+    """
+    if not pipeline._py_graph_built:
+        pipeline._build_graph()
+    input_node_names = []
+    for op in pipeline._ops:
+        if isinstance(op._op, ExternalSource):
+            input_node_names.append(op.name)
+    return input_node_names
```

Comment:
This doesn't work with a deserialized pipeline - is that intentional (or at least known) limitation?